src/oidc/oidc.util.ts

```diff
@@ -0,0 +1,190 @@
+
+let Provider, oidc;
+const configuration: Configuration = {
+  // ... see the available options in Configuration options section
+  features: {
+    // claimsParameter:{
+    //     enabled: false
+    // },
+
+    /* enables grant_type=client_credentials to be used on token endpoint */
+    clientCredentials: {
+      enabled: true,
+    },
+    resourceIndicators:{
+      enabled: true,
+      getResourceServerInfo(ctx,resourceIndicator,client){
+        if(resourceIndicator)
+          return {
+            scope: 'openid email profile phone offline_access address',
+            audience: resourceIndicator,
+            accessTokenTTL: 1*60*60,
+            accessTokenFormat: 'jwt'
+          }
+        throw new Error("LOL")
+      }
+    },
+    /* To extract out the info in the tokens generated, default has to be modified to allow introspection from authentic sources only (features.introspection.allowedPolicy)*/
+    introspection: {
+      enabled: true,
+      // this is default function, change it
+      // allowedPolicy(ctx, client, token) {
+      //   if (client.clientAuthMethod === 'none' && token.clientId !== ctx.oidc.client.clientId) {
+      //     return false;
+      //   }
+      //   return true;
+      // },
+    },
+    /* resourceIndicators: {
+       enabled: true,
+        getResourceServerInfo(ctx, resourceIndicator) {
+        if (resourceIndicator === 'urn:api') {
+           return {
+             scope: 'read',
+             audience: 'urn:api',
+             accessTokenTTL: 1 * 60 * 60, // 1 hour
+             accessTokenFormat: 'jwt',
+           };
+         }
+
+         throw new Error('Invalid target');
+       },
+     }, */
+    /* To turn off default pages and routes */
+    devInteractions: {
+      enabled: false,
+    },
+  },
+  interactions: {
+    url(_, interaction) {
+      return `/interaction/${interaction.uid}`;
+    },
+  },
+  clientBasedCORS(ctx, origin, client) {
+    return false;
+  },
+
+  //   claims:{
+  //     profile: ['username','gender','birthdate','email'],
+  //   },
+
+  clients: [
+    {
+      client_id: process.env.CLIENT_ID,
+      client_secret: process.env.CLIENT_SECRET,
+      redirect_uris: [
+        `${process.env.HOST_NAME}:${process.env.HOST_PORT}/${process.env.OIDC_CALLBACK_ROUTE}`,
+      ],
+      // + other client properties
+      grant_types: [
+        'authorization_code',
+        'client_credentials',
+        'refresh_token',
+      ],
+      //   redirect_uris: [],
+      response_types: ['code'],
+    },
+  ],
+  pkce: {
+    methods: ['S256'],
+    required: () => false,
+  },
+  scopes: ['openid', 'offline_access', 'profile', 'email', 'phone', 'address'],
+
+  /*   findAccount: async function findAccount(ctx,sub,token){
+
+  //   }
+
+  //   interactions: {
+  //     url(ctx, interaction) {
+  //       // eslint-disable-line no-unused-vars
+  //       return `/interaction/${interaction.uid}`;
+  //     },
+  //   }, */
+
```

Comment:
remove unless this is for document purposes, if this is for document purposes, change this to use JSDoc format.